src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -1304,18 +1304,17 @@ public static unsafe SocketError AcceptAsync(Socket socket, SafeSocketHandle han
             try
             {
                 // This can throw ObjectDisposedException.
-                int bytesTransferred;
                 bool success = socket.AcceptEx(
                     handle,
                     acceptHandle,
                     Marshal.UnsafeAddrOfPinnedArrayElement(asyncResult.Buffer!, 0),
                     receiveSize,
                     addressBufferSize,
                     addressBufferSize,
-                    out bytesTransferred,
+                    out int bytesReceived,
                     asyncResult.DangerousOverlappedPointer); // SafeHandle was just created in SetUnmanagedStructures
 
-                return asyncResult.ProcessOverlappedResult(success, 0);
+                return asyncResult.ProcessOverlappedResult(success, bytesReceived);
```

Comment:
Was this an error? The original `0` passed to `ProcessOverlappedResult`?

Reply:
Yes, if I understand the code here correctly, `BeginAccept(receiveSize)` / `EndAccept(out buffer)` combo will report receiving 0 bytes in the synchronous completion case.
For async completions, the BytesTransfered is correctly passed to the AsyncResult CompletionCallback.

Reply:
Looks like the Unix implementation just does a `Debug.Assert(receiveSize == 0, $"Unexpected receiveSize: {receiveSize}");` and never uses the value (we don't actually guard against non-zero values being passed in by the user).

So the overload is useless on Unix and wrong on Windows for synchronous completions.
I will revert this part of the change as it doesn't really fit into this PR.